test/integration/ExamplesBuild_TEST.cc

```diff
@@ -0,0 +1,414 @@
+
+//////////////////////////////////////////////////
+bool process_status_failure()
+{
+  int errval(::GetLastError());
+
+  if (not_found_error(errval))
+  {
+    return false;
+  }
+  else if ((errval == ERROR_SHARING_VIOLATION))
+  {
+    return true;  // odd, but this is what boost does
+  }
+  return false;
+}
+
+//////////////////////////////////////////////////
+bool internal_check_path(const std::string &_path, DWORD &attr)
+{
+  attr = ::GetFileAttributesA(_path.c_str());
+  if (attr == 0xFFFFFFFF)
+  {
+    return process_status_failure();
+  }
+
+  // reparse point handling;
+  //   since GetFileAttributesW does not resolve symlinks, try to open file
+  //   handle to discover if the file exists
+  if (attr & FILE_ATTRIBUTE_REPARSE_POINT)
+  {
+    handle_wrapper h(
+        create_file_handle(
+          _path,
+          0,  // dwDesiredAccess; attributes only
+          FILE_SHARE_DELETE | FILE_SHARE_READ | FILE_SHARE_WRITE,
+          0,  // lpSecurityAttributes
+          OPEN_EXISTING,
+          FILE_FLAG_BACKUP_SEMANTICS,
+          0));  // hTemplateFile
+    if (h.handle == INVALID_HANDLE_VALUE)
+    {
+      return process_status_failure();
+    }
+
+    if (!is_reparse_point_a_symlink(_path))
+    {
+      return true;
+    }
+  }
+
+  return true;
+}
+#endif
+
+//////////////////////////////////////////////////
+bool isDirectory(const std::string &_path)
+{
+#ifndef _WIN32
+  struct stat path_stat;
+
+  if (::stat(_path.c_str(), &path_stat) != 0)
+  {
+    return false;
+  }
+
+  // cppcheck-suppress ConfigurationNotChecked
+  return S_ISDIR(path_stat.st_mode);
+#else
+  DWORD attr;
+
+  if (internal_check_path(_path, attr))
+  {
+    return (attr & FILE_ATTRIBUTE_DIRECTORY) != 0;
+  }
+
+  return false;
+#endif
+}
+
+//////////////////////////////////////////////////
+bool isFile(const std::string &_path)
+{
+  std::ifstream f(_path);
+  return (!isDirectory(_path)) && f.good();
+}
+
+//////////////////////////////////////////////////
+void removeAll(const std::string &_path)
+{
+  if (isDirectory(_path))
+  {
+    DIR *dir = opendir(_path.c_str());
+    if (dir)
+    {
+      struct dirent *p;
+      while ((p=readdir(dir)))
+      {
+        // Skip special files.
+        if (!std::strcmp(p->d_name, ".") || !std::strcmp(p->d_name, ".."))
+          continue;
+
+        removeAll(_path + "/" + p->d_name);
+      }
+    }
+    closedir(dir);
+
+    // Remove the directory
+    bool removed;
+#ifdef _WIN32
+    removed = RemoveDirectory(_path.c_str());
+#else
+    removed = (rmdir(_path.c_str()) == 0);
+    if (!removed)
+    {
+      // A sym link would end up here
+      removed = (std::remove(_path.c_str()) == 0);
+    }
+#endif
+    ASSERT_TRUE(removed);
+  }
+  else if (isFile(_path))
+  {
+    const bool removed = (std::remove(_path.c_str()) == 0);
+    ASSERT_TRUE(removed);
+  }
+}
+
+//////////////////////////////////////////////////
+TEST(ExamplesBuild, Build)
+{
+  // \todo(nkoenig) Fix windows.
+#ifndef _WIN32
```

Comment:
Should we add all this windows stuff if we're going to skip it all? I think it's better to leave it out, and bring it in when we know it works on Windows. Otherwise this is all currently dead code.